### PR TITLE
Upgrade Ubuntu to latest LTS (Bionic 18.04)

### PR DESCRIPTION
**Pull request by BradenAtSiemens**
For your consideration, an upgrade of Ubuntu to its latest LTS (Bionic 18.04).

### CODE REVIEW

**Comment by parente** [comment]
> Hmm... it seems like hkp://p80.pool.sks-keyservers.net:80 occationally cannot be contacted, though using that key server is not part of this change...

Did you experience build failures because of this? /cc @taqtiqa-mark

File: scipy-notebook/Dockerfile [hunk 1]
```diff
@@ -6,9 +6,9 @@ LABEL maintainer="Jupyter Project <jupyter@googlegroups.com>"
 
 USER root
 
-# libav-tools for matplotlib anim
+# ffmpeg for matplotlib anim
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends libav-tools && \
+    apt-get install -y --no-install-recommends ffmpeg && \
```
Comment (parente) [inline scipy-notebook/Dockerfile]: Noting this resolves https://github.com/jupyter/docker-stacks/issues/626

**Comment by parente** [comment]
I'm 👍 on this since 18.04 is a LTS release. 

I don't have a way to test the older mesos client compatibility, but we're discussing how to make the Spark images more cluster agnostic in #626 anyway.

I'd like to leave this open for a few days to see if anyone has reservations. Then we can ship it.

File: scipy-notebook/Dockerfile (hunk 1 above)
Comment (taqtiqa-mark) [reply scipy-notebook/Dockerfile]: Think this resolves #611 rather than #626

**Comment by taqtiqa-mark** [comment]
Yes, if the consensus/decision is to make the `all-spark-notebook` generic then the current Mesos container would be the result of a  [buildah script](https://github.com/taqtiqa-mark/bob) or a Docker [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds) such as:

````
FROM jupyter/all-spark-notebook
FROM mesosphere/mesos
````

File: scipy-notebook/Dockerfile (hunk 1 above)
Comment (parente) [reply scipy-notebook/Dockerfile]: Indeed. Wrong #

**Comment by BradenAtSiemens** [comment]
Hmm... looking more carefully at the build log, for some reason in the notebooks that build off of scipy-notebook, `blas` gets downgraded from 1.1 from the `conda-forge` repo to 1.0 from the `defaults` repo, which then seems to trigger a downgrade of `numpy`, `scipy`, and `scikit-learn`...

If I pin the install of `blas` to `1.1*` (currently it is just `*`), it seems to avoid this issue, so I'll add a commit for that.

**Comment by parente** [comment]
Nice catch. I don't see any other concerning downgrades in the latest build log. Let's get this merged.

Thanks for the PR and the testing! 🍰

**Comment by jakirkham** [comment]
@BradenAtSiemens, have also been running into some annoyances with `blas` downgrades recently and know of others struggling with this issue as well. Think we may want to use `conda`'s MatchSpec feature to pin `blas` to `conda-forge`. Believe this is a consequence of changes in `defaults` that are affecting us currently, which this proposed change would workaround. Will get a PR going.

cc @kalefranz @msarahan (for awareness)

**Comment by jakirkham** [comment]
Have placed this change in PR ( https://github.com/jupyter/docker-stacks/pull/649 ).

**Comment by Bidek56** [comment]
Mesos build is not available for 18.04, why would you hardcode the Mesos repo to xenial? Do you know if it works on 18.04?

**Comment by Bidek56** [comment]
According to [MESOS-8921](https://issues.apache.org/jira/browse/MESOS-8921l) Mesos does NOT work with 18.04.

**Comment by taqtiqa-mark** [comment]
Maybe this is a natural point at which to start to disentangle the Spark Cluster Manager (SCM) and spark from the containers?

**Comment by taqtiqa-mark** [comment]
Maybe 18.04 is the natural point at which to remove the dependency on the version of Spark and its Cluster Manager?

**Pull request merged**